src/basic/rlimit-util.c

```diff
@@ -13,6 +13,10 @@
 #include "string-table.h"
 #include "time-util.h"
 
+static inline int rlimit_to_nice(rlim_t prio) {
```

Comment:
please drop the "inline". The compiler can determine on its own if it wants to inline this or not.

Using "inline" makes sense if used in header files, but not in .c files.

Reply:
@poettering sorry to raise this from the dead, but why?

I was just looking at callgrind of `journalctl --verify` here and to my surprise siphash24.c::rotate_left() is at the top of the profile, being called as a distinct function for some insane reason.

What's the reason to not instruct inline?  should we go back to using macros for such hot one-liners we *never* want to incur a function call for, if static inline is somehow frowned upon?

Reply:
The compiler can ignore the inline anyway. The compiler should be better to figure out when to inline something than we are. it should inline stuff where it makes sense that isn't marked for inlining and vice versa should be fine with not inlining stuff that it doesn't want to.

One can force inlining with that always_inline gcc attribute, but it's really not clear to me when that is a good idea.

We dropped all "inline" uses from our tree, to let the compiler figure this out on its own. We only use it in headers now, where it is necessary if we want to write code into headers. Unles LTO is used the compiler is not able to inline exported stuff, so for such exported stuff we need to declare inline manually. But for local stuff: let the compiler figure it out.

Macros are not the solution: the compiler should inline on its own.

maybe your built was without the right optimization settings? i.e. unoptimized builds will mean no inlining is done?

Also, note that inlining can be expensive, since it increases code size in many cases, and that increases cache pressure, which makes stuff slower. it's really hard to know in advance when inlining is benefitial performance-wise and size-wise, which is why we want to leave this to the compiler, which should be able to estimate this way better than us.

No idea why the roatet_left became so slow. Smells like a compiler parameters issue (or a compiler bug).

We could add:

```c
#ifdef __i386__
#include <ia32intrin.h>
#endif
```

and then use the `_rotl` intrinsic on x86 platforms. That will force to code be a single opcode on x86 at least